python/ccxt/binance.py

```diff
@@ -1557,15 +1558,22 @@ def fetch_trades(self, symbol, since=None, limit=None, params={}):
     def parse_order_status(self, status):
         return self.safe_string(STATUSES_MAPPING, status, status)
 
-    def parse_trades_cost_fee(self, trades):
-        cost = trades[0]['cost']
-        fee = {
-            'cost': trades[0]['fee']['cost'],
-            'currency': trades[0]['fee']['currency'],
-        }
-        for i in range(1, len(trades)):
-            cost = self.sum(cost, trades[i]['cost'])
-            fee['cost'] = self.sum(fee['cost'], trades[i]['fee']['cost'])
+    def parse_trades_cost_fee(self, symbol, trades):
+        cost, fees, fee = 0., defaultdict(lambda: {'cost': 0.}), None
+        for trade in trades:
+            cost += trade['cost']
+            trade_fee = trade['fee']
+            _fee_cost = trade_fee['cost']
+            _fee_currency = trade_fee['currency']
+            fees[_fee_currency]['currency'] = _fee_currency
+            fees[_fee_currency]['cost'] += _fee_cost
```

Comment:
I think we better preserve the usage of self.sum just be more "Secure".
also in line 1563